File: app/mobile_app/lib/models/like_toggle_result.dart
import 'package:json_annotation/json_annotation.dart';

part 'like_toggle_result.g.dart';

@JsonSerializable()
class LikeToggleResult {
  final bool isLiked;
  final bool isDisliked;
  final int likeCount;
  final int dislikeCount;

  LikeToggleResult({
    required this.isLiked,
    required this.isDisliked,
    required this.likeCount,
    required this.dislikeCount,
  });

  factory LikeToggleResult.fromJson(Map<String, dynamic> json) =>
      _$LikeToggleResultFromJson(json);
  Map<String, dynamic> toJson() => _$LikeToggleResultToJson(this);
}

File: api/MarketOurs.WebAPI/MarketOurs.DataAPI/Services/LikeToggleResult.cs
namespace MarketOurs.DataAPI.Services;

/// <summary>
/// 点赞/点踩切换操作的结果
/// </summary>
/// <param name="IsLiked">操作后用户是否点赞了该目标</param>
/// <param name="IsDisliked">操作后用户是否点踩了该目标</param>
/// <param name="LikeCount">操作后的点赞总数</param>
/// <param name="DislikeCount">操作后的点踩总数</param>
public record LikeToggleResult(bool IsLiked, bool IsDisliked, int LikeCount, int DislikeCount);
